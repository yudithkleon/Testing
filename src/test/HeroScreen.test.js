import { mount } from "enzyme";
import React from "react";
import { MemoryRouter, Route } from "react-router";
import { HeroScreen } from "../components/heroes/HeroScreen";
import { getHeroById } from "../selectors/getHeroById";



describe("Prueba del componente <HeroScreen", () => {

//   const hero = useMemo(() => getHeroById(heroeId), [heroeId]);
//   console.log("heroe: ", +hero);

  const historyMock = {
    push: jest.fn(),
    goBack: jest.fn(),
    length: 10,
  };
  const wrapper = mount(
    <MemoryRouter initialEntries={["/Hero"]}>
      <HeroScreen history={historyMock} />
    </MemoryRouter>
  );

  test("mostrar el redirect si no tenemos URL del heroe", () => {
    expect(wrapper.find("Redirect").exists()).toBe(true);
  });

  test("El componente debe de funcionan correctamente", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/hero/marvel-hulk"]}>
        <Route parth="/hero/:heroeId" component={HeroScreen} />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('Hacer push', ()=>{
      const historyMock ={
          push: jest.fn(),
          goBack: jest.fn(),
          length:1
      }
      const wrapper = mount(
        <MemoryRouter initialEntries={["/hero/marvel-hulk"]}>
          <Route parth="/hero/:heroeId" component={()=> <HeroScreen history={historyMock}/>} />
        </MemoryRouter>
      );
        wrapper.find('button').props('onClick')();
        expect(historyMock.push).toHaveBeenCalledWith('/')
        expect(historyMock.goBack).not.toHaveBeenCalled();
  })

  test ('regresar a la pantalla anterior', ()=>{
    const historyMock ={
        push: jest.fn(),
        goBack: jest.fn(),
        length: 10
    }
    const wrapper = mount(
      <MemoryRouter initialEntries={["/hero/marvel-hulk"]}>
        <Route parth="/hero/:heroeId" component={()=> <HeroScreen history={historyMock}/>} />
      </MemoryRouter>
    );
    wrapper.find('button').props('onClick')();
    expect(historyMock.push).toHaveBeenCalledWith('/')
    expect(historyMock.goBack).toHaveBeenCalled();
  })

});
