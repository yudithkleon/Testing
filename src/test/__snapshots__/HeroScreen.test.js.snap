// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente <HeroScreen El componente debe de funcionan correctamente 1`] = `""`;
